assets/src/dashboard/app/views/shared/storyGridView.js

```diff
@@ -116,45 +119,47 @@ const StoryGridView = ({
   );
 
   return (
-    <StoryGrid>
-      {filteredStories.map((story) => (
-        <CardGridItem key={story.id} isTemplate={isTemplate}>
-          <CardPreviewContainer
-            centerAction={{
-              targetAction: story.centerTargetAction,
-              label: centerActionLabel,
-            }}
-            bottomAction={{
-              targetAction: story.bottomTargetAction,
-              label: bottomActionLabel,
-            }}
-          >
-            <PreviewErrorBoundary>
-              <PreviewPage page={story.pages[0]} />
-            </PreviewErrorBoundary>
-          </CardPreviewContainer>
-          {!isTemplate && (
-            <DetailRow>
-              <CardTitle
-                title={story.title}
-                modifiedDate={story.modified.startOf('day').fromNow()}
-                onEditComplete={(newTitle) =>
-                  handleOnRenameStory(story, newTitle)
-                }
-                onEditCancel={() => setTitleRenameId(-1)}
-                editMode={titleRenameId === story.id}
-              />
-              <CardItemMenu
-                onMoreButtonSelected={setContextMenuId}
-                contextMenuId={contextMenuId}
-                onMenuItemSelected={handleMenuItemSelected}
-                story={story}
-              />
-            </DetailRow>
-          )}
-        </CardGridItem>
-      ))}
-    </StoryGrid>
+    <UnitsProvider pageSize={pageSize}>
```

Comment:
Oh no, I meant to keep `UnitsProvider` in the `details.js` file but wrap `StoryGridView` component with it.  I only say this because `StoryGridView` is a shared component with the the myStories index and it has its own `UnitsProvider` wrapped with it.  Generally we want to keep our shared components context and provider free and let the parent handle the data.

`CardGallery` is special because it needs two different `UnitProviders` (one for the mini-card and another for the main card) so I need it to host the provider itself, but for every other situation, it's best to keep it at the parent level... sorry for the confusion 😱

Reply:
ahhhh i had misunderstood this thinking that if we moved the unitsProvider to the gridview it would take care of it without needing to wrap it on each view, total oversight on my part! Fixing!